### ISSUE TITLE
Cannot compile autogenerated code because of multiple java errors

### ISSUE THREAD

**Issue by saibotma** [jOOQ/jOOQ#7684]
I used `java --add-modules java.xml.bind -classpath jooq-3.11.3.jar;jooq-meta-3.11.3.jar;jooq-codegen-3.11.3.jar;postgresql-42.2.4.jar;. org.jooq.codegen.GenerationTool jooq.xml` to autogenerate JOOQ code for my PostgreSQL 10 `USER` table.
![pgadmin4_2018-07-24_08-08-49](https://user-images.githubusercontent.com/26022902/43119986-e4e88868-8f18-11e8-95a0-3a8f5301c3c8.png)

The codegen tool finishes successfully however my program won´t compile because there are several Java syntax errors in the autogenerated code. 

Some examples:
`PgClass.java`
```
/**
* @deprecated Unknown data type. Please define an explicit {@link org.jooq.Binding} to specify how this type should be handled. Deprecation can be turned off using <deprecationOnUnknownTypes/> in your code generator configuration.
*/
@java.lang.Deprecated
public final TableField<PgClassRecord, Object> RELPARTBOUND = createField("relpartbound", , this, "");
```

![idea64_2018-07-24_08-12-48](https://user-images.githubusercontent.com/26022902/43120125-624f401c-8f19-11e8-8224-ce7bd4cbc277.png)

The compiler is telling me `java: illigal start of expression`

`PgIndex.java:`
```
/**
* The column <code>pg_catalog.pg_index.indoption</code>.
*/
public final TableField<PgIndexRecord, Object[]> INDOPTION = createField("indoption", .getArrayDataType(), this, "");
```
![idea64_2018-07-24_08-15-31](https://user-images.githubusercontent.com/26022902/43120224-bec0dedc-8f19-11e8-9b27-7fd83520a197.png)

The compiler is telling me `java: as of release 8, 'this' is allowed as the parameter name for the receiver type only, which has to be the first parameter`

- jOOQ: 3.11.3
- Java: 1.8.0_181
- Database (include vendor): PostgreSQL 10
- OS: Windows 10

**Comment by saibotma** [comment]
Fixed the issue: https://stackoverflow.com/questions/51494099/cannot-compile-autogenerated-code-because-of-multiple-java-errors

**Comment by lukaseder** [comment]
This issue is fixed through https://github.com/jOOQ/jOOQ/issues/7692

**Comment by lukaseder** [comment]
Thanks for documenting this problem!

**Comment by lukaseder** [comment]
@saibotma The fix for issue #7692 has been backported to version 3.11.4

**Comment by saibotma** [comment]
Thanks! I appreciate that :)
